x/mint/abci_app.go

```diff
@@ -6,21 +6,26 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
-// Called every block, process inflation on the first block of every hour
+// Inflate every block, update inflation parameters once per hour
 func BeginBlocker(ctx sdk.Context, k Keeper) {
 
 	blockTime := ctx.BlockHeader().Time
 	minter := k.GetMinter(ctx)
-	if blockTime.Sub(minter.InflationLastTime) < time.Hour { // only mint on the hour!
+	params := k.GetParams(ctx)
+
+	mintedCoin := minter.BlockProvision(params)
+	k.fck.AddCollectedFees(ctx, sdk.Coins{mintedCoin})
```

Comment:
Will we need decimal coins once we have "micro" denominations?

Reply:
not for nano-atoms if that what you mean. But we will still need dec_coins for the distribution mechanism because a 1-nano atom fee needs to be able to get divided and accumulated by many delegators.

Reply:
Alternatively, do we need micro denominations, when we could just do everything as DecCoins?

Reply:
How does making all coins DecCoins change anything? Now we just have to deal with two types, and all accounts have DecCoins?

However, we can get rid of DecCoins by making the nano-atom threshold sufficiently small such that all gas payments should still be a large number of the base unit, and we are fine with the minor truncations of total reward. (e.g. gas costs 1 million base unit)